### ISSUE TITLE
ai Support seems degraded starting between 7.1-0-10 and 7.1-0-16

### ISSUE THREAD

**Issue by brassel** [ImageMagick/ImageMagick#5658]
### ImageMagick version

7.1-0-16 and above

### Operating system

Other (enter below)

### Operating system, version and so on

docker using linux/amd64

### Description

For the attached file `seal_original.ai` the `convert`command for versions 7.1-0-16 and up will produce the non-optimal result attached as `seal_1_0_16.ai.png`. In versions 7.1-0-10 and at least some below it produced the expected result attached as `seal_1_0_10.ai.png`.
It will probably be of note that the corresponding `identify` commands disagree on the color format of the file, one reporting cmyk (versions 7.1-0-16 and up) while the older versions report srgba. 

### Steps to Reproduce

These are the commands I used:
~~~
convert -quiet -density 300 "seal_original.ai"'['0']' result.png
~~~
And in older versions
~~~
identify seal_original.ai
seal_original.ai AI 160x228 160x228+0+0 16-bit sRGB 2608B 0.030u 0:00.034
~~~
while in newer versions
~~~
identify seal_original.ai
seal_original.ai AI 160x228 160x228+0+0 16-bit ColorSeparation CMYK 193B 0.060u 0:00.045
~~~

### Images

![seal_1_0_10 ai png](https://user-images.githubusercontent.com/103310/195618974-679abe5b-6ea8-4cf3-b2ee-a15b2a40742f.png)
![seal_1_0_16 ai png](https://user-images.githubusercontent.com/103310/195618978-220a3361-371b-4a26-833b-4a30ae1823c2.png)
[seal_original.ai.zip](https://github.com/ImageMagick/ImageMagick/files/9777559/seal_original.ai.zip)

**Comment by snibgo** [comment]
> ... will produce the non-optimal result attached ...

What does "non-optimal" mean? What is the difference between results?

IM delegates the rasterization of AI files to Ghostscript (GS). Perhaps the difference you are seeing is from a change in GS. What versions of GS are you using?

**Comment by brassel** [comment]
Hi @snibgo, thanks for replying so fast.
Non-optimal means here that both the cmyk colors and the alpha channel were ignored.
And thank you for the hint about ghost script. I will inspect the Docker images and try to find out about the respective version.

**Comment by snibgo** [comment]
We can tell the rasterizer to create the image as sRGB, including alpha, with "-colorspace sRGB" *before* reading the input:
```
magick -colorspace sRGB seal_original.ai x.png
```

**Comment by brassel** [comment]
Hi @snibgo , 
thank you for the information! Indeed, this solves the problem with the first file. 
I still think that the output of `identify` in the different versions is suspicious, once `sRGB` and once `CMYK` don't you think? 
The file stems from a bigger test suite I ran end of last year and there are more files which do ignore CMYK now but have been correctly converted back then. These also feature a great difference with regard to `identify`. I have attached one example which - I hope - does not rely on Ghostscript as it is a raster TIF image.

![blister_2021 tif](https://user-images.githubusercontent.com/103310/195860405-b5341559-e24b-49e2-847c-5984d6d82bb8.png)
![blister_2022 tif](https://user-images.githubusercontent.com/103310/195860420-a96a2bce-c226-4ce4-b8fd-572bc9289f09.png)
[blister_original.tif.zip](https://github.com/ImageMagick/ImageMagick/files/9786783/blister_original.tif.zip)

Back end of last year the file was converted with the equivalent (I have just shortened the file paths) of the command 
~~~
convert -quiet "blister_original.tif"'['0']' -profile sRGB_profile.icc -define png:color-type=6 "blister_2021.png"
~~~
where `sRGB_profile.icc` is a standard sRGB profile which I add just for completeness.
 
[sRGB_profile.icc.zip](https://github.com/ImageMagick/ImageMagick/files/9786849/sRGB_profile.icc.zip)

But now the same command yields the result `blister_2022.png` which does not correctly convert CMYK as can be well seen in the blue and green areas in and around the "Alpenfest" sticker.

In 2021 the command 
~~~
identify -quiet -format 'Channel:%[channels],ClippingPath:%[8BIM:2999,2999:#1],Profile:%[profile:icc]\n' blister_original.tif
~~~
yielded
~~~
Channel:cmyka,ClippingPath:product,Profile:ISO Web Coated 0404
~~~
whereas now it only yields
~~~~
Channel:cmyka,ClippingPath:product,Profile:
~~~~
Thusly, the new versions do not detect neither the clipping path nor the color profile embedded in the TIF image.
Is there anything I can do or extend about the imagemagick installation such that it gets back its old power?

**Comment by brassel** [comment]
You also asked about the version of `ghostscript`, @snibgo.
On both images, `dpokidov/imagemagick:7.1.0-10` and `dpokidov/imagemagick:7.1.0-16` there is the same version `9.27`.
~~~~

# ghostscript --version
9.27
~~~~

**Comment by snibgo** [comment]
> I still think that the output of identify in the different versions is suspicious, once sRGB and once CMYK don't you think?

I don't know exactly what changed, or when, or why. Perhaps the old IM version didn't give users the choice of rasterizing colorspace, and always rasterized to sRGB. And perhaps this changed to leave the choice to GS unless the user specified a colorspace before the rasterizing.

> I have attached one example ...

When you show multiple images and multiple commands, please label them so we can easily see which command produced which image. I think you are saying the same command made both images, the first is recent, and the second was made in 2021. And you prefer the second version.

Your blister_original.tif is encoded as CMYK, with embedded ICC colour profile. The most accurate conversion to sRGB is by using "-profile", as you did. This assumes the embedded profile is correct.

CMYK colours may be outside the gamut of sRGB. Different strategies can be used to deal with this, with "-intent" before the "-profile", making up to four slightly different versions. I can't reproduce your second image with your supplied sRGB_profile.icc and any intent.

However, this command ...
```
magick blister_original.tif -colorspace sRGB -strip out.png
```
... makes a result that is the same as your second image. Not *roughly* the same, but *exactly* the same. So I suppose this is what you did in 2021. This conversion is technically wrong, as it ignores the embedded profile and uses a simple mathematical transformation from the CMYK channels to RGB.

With this command...
```
magick identify -quiet -format 'Channel:%[channels],ClippingPath:%[8BIM:2999,2999:#1],Profile:%[profile:icc]\n' blister_original.tif

'Channel:cmyka,ClippingPath:product,Profile:ISO Web Coated 0404
```
... with IM v7.1.0-49, I get the name of the embedded profile.

**Comment by brassel** [comment]
Hi @snibgo, thank you very much for this extremely helpful input!

> When you show multiple images and multiple commands, please label them so we can easily see which command produced which image. [...] And you prefer the second version.

I will keep that in mind, thank you! And I have updated the description hopefully improving it. With the updated labels I hope that it is clear that I do not prefer the second but the first option.

> CMYK colours may be outside the gamut of sRGB. Different strategies can be used to deal with this, with "-intent" before the "-profile", making up to four slightly different versions. I can't reproduce your second image with your supplied sRGB_profile.icc and any intent.

This comment was really enlightening. Using any of the 4 intents I found when researching your proposal has brought satisfying results, thank you once more. But the sad truth I found out, is that the results only look good when I use the imagemagick installed directly on my mac. If I use any of the many docker images I downloaded for testing, the result looks like the bad option `blister_2022.png`. So for some reason using imagemagick over docker was the root of my problem. 

The next thing I tried is that instead of using `--mount type=bind` to inject my local image files I used the volume option `-v`. And - voilà - the results suddenly looked as they should. How the way to access the local file system can change the results of conversion is beyond me. However, this case has been solved.

I am very sorry to have wasted your time like this. 
Is there a coffee jar where I could buy you a drink to make up for that?
